lib/src/main/kotlin/bstrees/balancers/RBBalancer.kt

```diff
@@ -3,8 +3,86 @@ package bstrees.balancers
 import bstrees.nodes.RBNode
 
 class RBBalancer<T : Comparable<T>> : TreeBalancer<T, RBNode<T>> {
-    override fun balanceAfterInsertion(node: RBNode<T>): RBNode<T> {
-        TODO("Not yet implemented")
+    private fun color(node: RBNode<T>?) = node?.color ?: RBNode.Color.Black
+
+    private fun isRed(node: RBNode<T>?) = color(node) == RBNode.Color.Red
+
+    private fun isBlack(node: RBNode<T>?) = !isRed(node)
+
+    private fun getParent(node: RBNode<T>) = node.parent
```

Comment:
What's the point of `getParent`?
It's basically the same as just writing `node.parent`